doc/sphinx/proof-engine/ltac.rst

```diff
@@ -1661,6 +1661,100 @@ Testing boolean expressions: guard
    .. exn:: Condition not satisfied.
       :undocumented:
 
+Checking properties of terms
+----------------------------
+
+Each of the following tactics acts as the identity if the check
+succeeds, and results in an error otherwise.
+
+.. tacn:: constr_eq @one_term @one_term
+
+   Checks whether its arguments are equal ignoring differences in alpha
+   conversion, casts and universe constraints. It may unify universes.
+
+   .. exn:: Not equal.
+      :undocumented:
+
+   .. exn:: Not equal (due to universes).
+      :undocumented:
+
+.. tacn:: constr_eq_strict @one_term @one_term
+
+   This tactic checks whether its arguments are equal modulo alpha
+   conversion, casts and universe constraints. It does not add new
+   constraints.
+
+   .. exn:: Not equal.
+      :undocumented:
+
+   .. exn:: Not equal (due to universes).
+      :undocumented:
+
+.. tacn:: constr_eq_nounivs @one_term @one_term
+
+   Checks whether its arguments are equal modulo alpha
+   conversion, casts and universe constraints. It does not add new
+   constraints.
+
+.. tacn:: unify @one_term @one_term {? with @ident }
+
+   Checks whether its arguments are unifiable, potentially
+   instantiating existential variables.
+
+   Unification takes the transparency information defined in the hint database
+   :n:`@ident` into account (see :ref:`the hints databases for auto and eauto
+   <thehintsdatabasesforautoandeauto>`).
+
+.. tacn:: is_evar @term
+   :name: is_evar
+
+   Checks whether its argument is a current existential
+   variable. Existential variables are uninstantiated variables generated
+   by :tacn:`eapply` and some other tactics.
+
+   .. exn:: Not an evar.
+      :undocumented:
+
+.. tacn:: has_evar @term
+   :name: has_evar
+
+   Checks whether its argument has an existential variable as
+   a subterm. Unlike context patterns combined with ``is_evar``, this tactic
+   scans all subterms, including those under binders.
+
+   .. exn:: No evars.
+      :undocumented:
+
+.. tacn:: is_var @term
+   :name: is_var
+
+   Checks whether its argument is a variable or hypothesis in
+   the current local context.
+
+   .. exn:: Not a variable or hypothesis.
+      :undocumented:
+
+.. tacn:: is_cofix @one_term
+   :undocumented:
+
+.. tacn:: is_const @one_term
+   :undocumented:
+
+.. tacn:: is_constructor @one_term
+   :undocumented:
+
+.. tacn:: is_fix @one_term
+   :undocumented:
```

Comment:
```suggestion
   :name: is_fix
   
   Checks whether its argument is a bare fixpoint.
   
   .. exn:: not a fix definition.
      :undocumented:
```

Here's an example, if you want to work it in:
```coq
Goal True.
  is_fix (fix f (n : nat) := match n with S n => f n | O => O end).
```

Reply:
"bare" meaning that the top node of the AST is a fix construct?

Reply:
Yes

Reply:
Can you confirm that a `let fix` will fail?  I'm adding a hyperlink to this nonterminal and if it does fail, I want to mention that `let fix` fails.

![image](https://user-images.githubusercontent.com/1253341/144501274-1653bd6c-fa78-444d-983f-3217f1b7cb21.png)

Reply:
`let (co)fix name body in term` is basically just syntactic sugar for `let name := (co)fix name body in term`, in the same way that `if` and destructuring `let` are just syntactic sugar for `match`.  Examples:
```coq
Goal True.
  is_fix (let fix f (n : nat) := match n with S n => f n | O => O end in f).
  (* Error: Tactic failure: not a fix definition. *)
```
```coq
Require Import Coq.Lists.Streams.
Goal True.
  is_cofix (let cofix f : Stream unit := Cons tt f in f).
  (* Error: Tactic failure: not a cofix definition. *)
```